app/src/ui/dialog/dialog.tsx

```diff
@@ -719,6 +719,25 @@ export class Dialog extends React.Component<DialogProps, IDialogState> {
       'tooltip-host'
     )
 
+    const ariaLabelledby = [
+      this.state.titleId,
+      // VoiceOver on macOS will not read the aria described by test thus we
+      // need to put it in the ariaLabeledBy
+      __DARWIN__ && this.props.role === 'alertdialog'
+        ? this.props.ariaDescribedBy
+        : null,
+    ].join(' ')
+
+    // VoiceOver on macOS will not read the aria-describedby thus we need to put
+    // it in the ariaLabeledBy to make sure it get's read-> which is only
+    // important on alertdialogs. This is bit hacky as semantically it should be
+    // in the aria-describedby.. maybe VoiceOver will be updated one day and we
+    // can just use the aria described by like we are supposed to.
+    const ariaDescribedBy =
+      __DARWIN__ && this.props.role === 'alertdialog'
+        ? ''
+        : this.props.ariaDescribedBy
```

Comment:
So... this is a bit hacky semantically. 

 But, is similar to the recommendation we got for popovers and is that we should use the `aria-describedby` and `aria-labelledby` combination that gets the job done. For confirm or error dialogs, the message should be announced to the user on dialog open. :/  Unfortunately, VoiceOver does not read the `aria-describedby` even for  dialogs with a role of `alertdialog` (this is the expected behavior of this markup). Thus, I added the `aria-describedby` id's into the `aria-labelledby` attribute only on macOS because NVDA reads as expected.